### ISSUE TITLE
report an error when a type variable cannot be resolved

### ISSUE THREAD

**Issue by KotlinIsland** [DetachHead/basedpyright#1499]
### Description

Code sample in [basedpyright playground](https://basedpyright.com/?typeCheckingMode=all&code=IYAgvCDaC6Q)

```python
a = []
```

### also:

this shouldn't report an error, it should be an error on the callsite
```py
class A[T]:
    def __init__(self, t: T = ...): ...
```

**Comment by DetachHead** [comment]
it's an error on the usage, but i agree it should probably also be an error on the declaration:

Code sample in [basedpyright playground](https://basedpyright.com/?typeCheckingMode=all&code=IYAgvCDaC6CwBQCAOAnAlgOwC4ApgEog)

```python
a = []

print(a) # error: reportUnknownArgumentType
```

as for the second example that seems unrelated? i don't see why it shouldn't be an error on the declaraion

**Comment by KotlinIsland** [comment]
> i don't see why it shouldn't be an error on the declaraion

pyi moment, but a better egg is `def f[T](*t: T)`, which is the same as list

**Comment by DetachHead** [comment]
#597?
